### PR TITLE
allow to download/preserve s3 folder structure

### PR DESCRIPTION
**Pull request by carebare47**
## Proposed changes

allow to download/preserve s3 folder structure

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Comment by rnzenha-s** [comment]
Could it be worth preventing it from re-downloading files that already exist locally?

File: sr_utilities_common/src/sr_utilities_common/aws_manager.py [hunk 1]
```diff
@@ -110,15 +111,33 @@ def _prepare_structure_download(self, files_base_path, files_folder_path, file_n
...
+                object_size = self._client.head_object(Bucket=bucket_name, Key=aws_path)['ContentLength']
+                with tqdm(total=object_size,
+                          unit="B", unit_scale=True,
+                          desc=file_full_path.replace(directory, ''),
+                          leave=False) as pbar:
+                    self._client.download_file(
+                        Bucket=bucket_name,
+                        Key=aws_path,
+                        Filename=file_full_path,
+                        Callback=lambda bytes_tx: pbar.update(bytes_tx),  # pylint: disable=W0640,W0108
```
Comment (niko-holmes) [inline sr_utilities_common/src/sr_utilities_common/aws_manager.py:139]: Why use a lamba function over `Callback=pbar.update`?

File: sr_utilities_common/src/sr_utilities_common/aws_manager.py [hunk 2]
```diff
@@ -147,16 +166,19 @@ def gather_all_files_local(files_base_path, files_folder_path):
...
+            print_msg_2 += f"\n    {files_base_path}/{files_folder_path}/{filename}"
         if bucket_subfolder:
             print_msg += f"\n    {bucket_subfolder}/{filename}"
         else:
             print_msg += f"\n    {filename}"
-    print_msg += f"\n\nThese files will be downloaded too {files_base_path}/{files_folder_path}:"
+    print_msg += f"\n\nThese files will be downloaded to {files_base_path}/{files_folder_path}:"
     print_msg += print_msg_2
     rospy.loginfo(print_msg)
     value = input("Would you like to download these files? (Y/N) ")
```
Comment (niko-holmes) [inline sr_utilities_common/src/sr_utilities_common/aws_manager.py]: Would suggest lowercase `(y/n)` instead of `(Y/N)`

**Review by niko-holmes — COMMENTED** [review]
Minor comments

File: sr_utilities_common/src/sr_utilities_common/aws_manager.py (hunk 2 above)
Comment (carebare47) [reply sr_utilities_common/src/sr_utilities_common/aws_manager.py]: that whole thing should be refactored, but done

**Review by niko-holmes — APPROVED** [review]

File: sr_utilities_common/src/sr_utilities_common/aws_manager.py (hunk 1 above)
Comment (dg-shadow) [reply sr_utilities_common/src/sr_utilities_common/aws_manager.py:139]: also interested in this question. appart tfrom the fact that lambdas are just neat

**Review by dg-shadow — APPROVED** [review]

**Pull request merged**